events/s3_batch_job.go

```diff
@@ -2,6 +2,18 @@
 
 package events
 
+// S3BatchJobResultCode represents the result of an S3BatchJobTask (i.e.
+// succeeded, permanent failure, or temmporary failure)
+type S3BatchJobResultCode string
+
+// S3BatchJobResultCode represents the result of an S3BatchJobTask (i.e.
+// succeeded, permanent failure, or temmporary failure)
+const (
+	S3BatchJobResultCodeSucceeded        S3BatchJobResultCode = "Succeeded"
```

Comment:
downside of changing `S3BatchJobResult.ResultCode` type back to `string`, is that comparisons against this new type require a cast!
```
if inputEvent.Results[0].ResultCode == S3BatchJobResultCodeSucceeded {
  // ..
}
```
-> 
```
Invalid operation: inputEvent.Results[0].ResultCode == S3BatchJobResultCodeSucceeded (mismatched types string and S3BatchJobResultCode)
```

Instead, go back to having S3BatchJobResult being a `S3BatchJobResultCode`. It is useful for self documentation purposes. But also change the type to be a type alias eg: `type S3BatchJobResultCoce = string` - this will preserve backwards compatibility.